digdag-standards/src/main/java/io/digdag/standards/operator/pg/PgConnection.java

```diff
@@ -0,0 +1,188 @@
+package io.digdag.standards.operator.pg;
+
+import java.util.UUID;
+import java.util.function.Consumer;
+import java.sql.Connection;
+import java.sql.Statement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.time.Duration;
+import com.google.common.base.Throwables;
+import io.digdag.client.config.Config;
+import io.digdag.util.DurationParam;
+import io.digdag.standards.operator.jdbc.AbstractJdbcConnection;
+import io.digdag.standards.operator.jdbc.AbstractPersistentTransactionHelper;
+import io.digdag.standards.operator.jdbc.DatabaseException;
+import io.digdag.standards.operator.jdbc.JdbcResultSet;
+import io.digdag.standards.operator.jdbc.TransactionHelper;
+import io.digdag.standards.operator.jdbc.NotReadOnlyException;
+import static java.util.Locale.ENGLISH;
+import static org.postgresql.core.Utils.escapeIdentifier;
+
+public class PgConnection
+    extends AbstractJdbcConnection
+{
+    protected static PgConnection open(PgConnectionConfig config)
+    {
+        return new PgConnection(config.openConnection());
+    }
+
+    protected PgConnection(Connection connection)
+    {
+        super(connection);
+    }
+
+    @Override
+    public void executeReadOnlyQuery(String sql, Consumer<JdbcResultSet> resultHandler)
+        throws NotReadOnlyException
+    {
+        try {
+            execute("SET TRANSACTION READ ONLY");
+            try (Statement stmt = connection.createStatement()) {
+                ResultSet rs = stmt.executeQuery(sql);  // executeQuery throws exception if given query includes multiple statements
+                resultHandler.accept(new PgResultSet(rs));
+            }
+            execute("SET TRANSACTION READ WRITE");
+        }
+        catch (SQLException ex) {
+            if (ex.getSQLState().equals("25006")) {  // 25006 = read_only_sql_transaction error
+                throw new NotReadOnlyException(ex);
+            }
+            else {
+                throw new DatabaseException("Failed to execute given SELECT statement", ex);
+            }
+        }
+    }
+
+    @Override
+    public String escapeIdent(String ident)
+    {
+        try {
+            StringBuilder buf = new StringBuilder();
+            escapeIdentifier(buf, ident);
+            return buf.toString();
+        }
+        catch (SQLException e) {
+            throw new IllegalArgumentException("Invalid identifier name (contains \\0 character): " + ident);
```

Comment:
According to [PostgreSQL JavaDoc](https://jdbc.postgresql.org/development/privateapi/org/postgresql/core/Utils.html#escapeIdentifier-java.lang.StringBuilder-java.lang.String-), escapeIdentifier throws SQLException only if the string contains a \0 character. Stacktrace is

```
irb(main):003:0* org.postgresql.core.Utils.escapeIdentifier(java.lang.StringBuilder.new, "ab\0").toString
Java::OrgPostgresqlUtil::PSQLException: Zero bytes may not occur in identifiers.
	from org.postgresql.core.Utils.doAppendEscapedIdentifier(org/postgresql/core/Utils.java:206)
	from org.postgresql.core.Utils.escapeIdentifier(org/postgresql/core/Utils.java:188)
	from java.lang.reflect.Method.invoke(java/lang/reflect/Method.java:498)
	from RUBY.<eval>((irb):3)
```

I think we don't have to include stacktrace.